lib/coverband/utils/file_path_helper.rb

```diff
@@ -0,0 +1,68 @@
+# frozen_string_literal: true
+
+####
+# Helper functions for shared logic related to file path manipulation
+####
+module Coverband
+  module Utils
+    module FilePathHelper
+      ###
+      # Takes a full path and converts to a relative path
+      ###
+      def full_path_to_relative(full_path)
+        relative_filename = full_path
+        root_paths.each do |root|
+          relative_filename = relative_filename.gsub(/^#{root}/, './')
+          # once we have a relative path break out of the loop
+          break if relative_filename.start_with? './'
+        end
+        relative_filename
+      end
+
+      ###
+      # relative_path_to_full code takes:
+      # relative_path: which is a full path the same as reported by Coverage
+      # roots: if a collection of all possible full app paths
+      #    EX: [Coverband.configuration.root_paths, "#{current_root}/"]
+      # The LAST item should be the current file system root
+      # it expands that expands and adds a '/' as that isn't there from Dir.pwd
+      #
+      # NOTEs on configuration.root_paths usage
+      # strings: matching is pretty simple for full string paths
+      # regex: to get regex to work for changing deploy directories
+      #        the regex must be double escaped in double quotes
+      #          (if using \d for example)
+      #        or use single qoutes
+      #        example: '/box/apps/app_name/releases/\d+/'
+      #        example: '/var/local/company/company.d/[0-9]*/'
+      ###
+      def relative_path_to_full(relative_path, roots)
+        relative_filename = relative_path
+        local_filename = relative_filename
+        roots.each do |root|
+          relative_filename = relative_filename.gsub(/^#{root}/, './')
+        end
+        # the filename for our reports is expected to be a full path.
+        # roots.last should be roots << current_root}/
+        # a fully expanded path of config.root
+        # filename = filename.gsub('./', roots.last)
+        # above only works for app files
+        # we need to rethink some of this logic
+        # gems aren't at project root and can have multiple locations
+        local_root = roots.find { |root| File.exist?(relative_filename.gsub('./', root)) }
+        local_root ? relative_filename.gsub('./', local_root) : local_filename
+      end
+
+      def root_paths
```

Comment:
I feel like this belongs on Coverband itself or the Coverband::Configuration and should be initialized once?

Reply:
yeah I thought about it as well I kept it here since it was all paths, but happy to move it there

Reply:
Makes sense. I can see that. I usually avoid modules for code reuse. Prefer to have methods that belong to objects for some reason. I find that going overboard with modules for organizing code can make things confusing and each one starts to feel like a junk drawer.

Reply:
I see your point about the junk drawer, but it doesn't feel this way to me. The helper is clearly named, the methods here are focused on paths and not something else. This feels like a modular codebase to me personally and quite a way off from being a junk drawer (although some junk drawers have their purpose too, but let's not go too far there...)

Reply:
For sure this is not a "junk drawer" and I probably shouldn't have used those words. I think in general this is a module with a bunch of helper methods that are similar in that they have to do file system stuff. This is a fine way to organize these methods. In general I prefer to have methods defined on objects if possible. If they are pure helper functions with no state and don't belong on an object, then I prefer making methods on the module object like:

```ruby
def self.relative_path_to_full(relative_path, roots)
```
This I find to be more self documenting regarding how this method is to be used..

```
FilePathHelper.relative_path_to_full('./foo/bar.rb', '/app/')
```

Hope that makes sense and in this case, I think it's fine to leave as is.

Reply:
OK moved two of the helpers to config and kept the others here for now.